server/norace_test.go

```diff
@@ -11252,3 +11252,59 @@ func TestNoRaceJetStreamClusterLargeMetaSnapshotTiming(t *testing.T) {
 	require_NoError(t, n.InstallSnapshot(snap))
 	t.Logf("Took %v to snap meta with size of %v\n", time.Since(start), friendlyBytes(len(snap)))
 }
+
+func TestNoRaceJetStreamClusterInfoOnMissingConsumers(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "R3F", 3)
+	defer c.shutdown()
+
+	nc, js := jsClientConnect(t, c.randomServer())
+	defer nc.Close()
+
+	// Create a stream just so the consumer info processing misses on the consumer only.
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:     "TEST",
+		Subjects: []string{"foo"},
+		Replicas: 3,
+	})
+	require_NoError(t, err)
+
+	done := make(chan bool)
+	pending := make(chan int, 1)
+
+	// Check to make sure we never have any pending on the API queue.
+	go func() {
+		ml := c.leader()
+		for {
+			select {
+			case <-done:
+				return
+			case <-time.After(100 * time.Millisecond):
+				if qlen := ml.jsAPIRoutedReqs.len(); qlen > 0 {
```

Comment:
May not be an issue but might be a bit more robust if it's `ml.jsAPIRoutedReqs.len() + ml.jsAPIRoutedReqs.inProgress()` as then it'll capture if there's a consumer info pulled off the queue and in progress. (The API worker doesn't `recycle()` until the ICB has finished.)

Reply:
ok will do. Noticed this though which felty weird.

`./norace_test.go:11282:13: invalid operation: ml.jsAPIRoutedReqs.len() + ml.jsAPIRoutedReqs.inProgress() (mismatched types int and int64)`